crates/core/src/storage/mod.rs

```diff
@@ -178,6 +397,21 @@ pub fn limit_store_handler<T: ObjectStore>(store: T, options: &StorageOptions) -
     }
 }
 
+/// Simple function to wrap the given [ObjectStore] in a [DeltaIOStorageBackend] if USE_EXPERIMENTAL_IO_RUNTIME is enabled
+pub fn deltars_store_handler(store: ObjectStoreRef, options: &StorageOptions) -> ObjectStoreRef {
+    let use_exerimental_rt = options
+        .0
+        .get(storage_constants::EXPERIMENTAL_RT)
+        .and_then(|v| v.parse::<bool>().ok())
+        .unwrap_or_default();
```

Comment:
One thing i wondered if this should really be done via the storage options, or if this should be configured via the `DeltaTableOptions`. What triggered this most, is thinking about stuff like coordinated commits or external catalogs, where other things may occur, that are io bound ...